src/backend/distributed/planner/distributed_planner.c

```diff
@@ -261,6 +264,15 @@ distributed_planner(Query *parse,
 												planContext.boundParams);
 			if (needsDistributedPlanning)
 			{
+				/*
+				 * Fast path queries cannot have merge command, and we
+				 * prevent the remaining here.
+				 */
+				ErrorIfQueryHasUnsupportedMergeCommand(planContext.originalQuery,
```

Comment:
I kind of missed that we added this line to this super important top-level function.

Can we move this to a more appropriate place? (e.g. router planner, and make other planners error for CMD_MERGE)

(the below call to TryToDelegateFunctionCall is already a quite ugly piece of this function, but at least it has some rationale behind it since it concerns a query that has no distributed tables)

Reply:
OK, we simply piggybacked on an existing check that was added as part of PG-15 support. In this phase, we are doing some of the deeper checks, such as updating-distribution-column further down in `ModifyQuerySupported->MergeQuerySupported`, it makes sense to converge them.